target/linux/ath79/dts/ar9132_buffalo_wzr-hp-g300nh.dts

```diff
@@ -0,0 +1,259 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9132.dtsi"
+
+/ {
+	compatible = "buffalo,wzr-hp-g300nh", "qca,ar9132";
+	model = "Buffalo WZR-HP-G300NH";
+
+	aliases {
+		led-boot = &led_diag;
+		led-failsafe = &led_security;
+		led-upgrade = &led_diag;
+	};
+
+	clock40mhz: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <40000000>;
+	};
+
+	gpio2: gpio-extender {
+		compatible = "nxp,74hc153-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+
+		// GPIOs used by this node
+		gpio-s0 = <&gpio 9 GPIO_ACTIVE_HIGH>;
+		gpio-s1 = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		gpio-1y = <&gpio 12 GPIO_ACTIVE_HIGH>;
+		gpio-2y = <&gpio 14 GPIO_ACTIVE_HIGH>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		aoss {
+			label = "aoss";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+/*
+		router_on {
+			label = "router_on";
+			linux,code = <BTN_5>;
+			gpios = <&gpio2 2 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		qos_off {
+			label = "qos_off";
+			linux,code = <BTN_3>;
+			gpios = <&gpio2 3 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+*/
+		usb {
+			label = "usb";
+			linux,code = <BTN_2>;
+			gpios = <&gpio2 5 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+/*
+		router_auto {
+			label = "router_auto";
+			linux,code = <BTN_6>;
+			gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+
+		qos_on {
+			label = "qos_on";
+			linux,code = <BTN_4>;
+			gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+			debounce-interval = <60>;
+		};
+*/
```

Comment:
I personally don't like adding so much commented code. I'd prefer to drop this from the DTS and add a comment about it in the commit message. When the problem is fixed, it should be relatively easy to insert this particular code again based on ar71xx.
Of course, the best solution would be to solve the underlying problem :-)